src/server/http/http.go

```diff
@@ -0,0 +1,172 @@
+package http
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/julienschmidt/httprouter"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
+)
+
+// HTTPPort specifies the port the server will listen on
+const HTTPPort = 652
+const apiVersion = "v1"
+
+func versionPath(p string) string {
+	return path.Join("/", apiVersion, p)
+}
+
+var (
+	getFilePath = versionPath("pfs/repos/:repoName/commits/:commitID/files/*filePath")
+	servicePath = versionPath("pps/services/:serviceName/*path")
+	loginPath   = versionPath("auth/login")
+	logoutPath  = versionPath("auth/logout")
+)
+
+type router = *httprouter.Router
+
+type server struct {
+	router
+	address        string
+	pachClient     *client.APIClient
+	pachClientOnce sync.Once
+	httpClient     *http.Client
+}
+
+// NewHTTPServer returns a Pachyderm HTTP server.
+func NewHTTPServer(address string) (http.Handler, error) {
+	router := httprouter.New()
+	s := &server{
+		router:     router,
+		address:    address,
+		httpClient: &http.Client{},
+	}
+
+	router.GET(getFilePath, s.getFileHandler)
+	router.POST(loginPath, s.authLoginHandler)
+	router.POST(logoutPath, s.authLogoutHandler)
+	router.GET(servicePath, s.serviceHandler)
+	router.POST(servicePath, s.serviceHandler)
+	// Debug method (to check login cookies):
+	router.GET(loginPath, s.loginForm)
```

Comment:
Since you're in here, we may as well remove this debug route. I haven't seen the need to use it since we got the login stuff working, so it can be removed (as well as the actually handler below).

Reply:
Will do.